lib/bike_brigade_web/live/campaign_live/campaign_message_list.ex

```diff
@@ -0,0 +1,50 @@
+defmodule BikeBrigadeWeb.CampaignLive.CampaignMessageList do
+  @moduledoc """
+  This module handles the logic for showing a list of chat message
+  with the riders for a specific campaign.
+  """
+  use BikeBrigadeWeb, :live_component
+  alias BikeBrigadeWeb.CampaignHelpers
+
+  alias BikeBrigade.LocalizedDateTime
+
+  alias BikeBrigade.Messaging
+  alias BikeBrigade.Delivery
+
+  import BikeBrigade.Utils, only: [humanized_task_count: 1]
+  import BikeBrigadeWeb.DeliveryHelpers
+  import BikeBrigadeWeb.Components.SMSMessageListComponent
+
+  alias BikeBrigadeWeb.Components.ConversationComponent
+
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    rider_ids =
+      assigns.riders
+      |> Map.values()
+      |> Enum.map(& &1.id)
+
+    conversations = Messaging.list_sms_for_campaign(rider_ids)
+
+    rider =
+      cond do
+        # TODO: why did I do this?
```

Comment:
I understand the code below but it may be clearer as a nested if or with some comments :)